File: Blind 75 LeetCode/Subhangi dhasmana/Reverse Bits/Reverse bits.py
class Solution:
    def reverseBits(self, n: int) -> int:
        def f(n,r,count):
            if n<1:
                return r<<(32-count)
            return f(n>>1,(r<<1)|(n&1),count+1)
        return f(n,0,0)

File: Blind 75 LeetCode/Abinand P/Reverse Bits/Reverse Bits.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t ans = 0;
        for(int i=0;i<32;i++){
            ans<<=1;
            ans|=(n&1);
            n>>=1;
        }
        return ans;
    }
};
